index.js

```diff
@@ -142,54 +156,169 @@ function updateSchemesAndHost(req) {
   }
 }
 
-module.exports.init = (aApp, aPredefinedSpec, aPath, aWriteInterval, aApiDocsPath = 'api-docs') => {
-  app = aApp;
-  predefinedSpec = aPredefinedSpec;
-  const writeInterval = aWriteInterval || 10 * 1000;
+/** TODO - type defs for Express don't work (I tried @external) */
+/**
+ * Apply this **first**!
+ *
+ * (straight after creating the express app (as the very first middleware))
+ *
+ * @description `response` middleware.
+ *
+ * @param {Express} expressApp - the express app
+ *
+ * @param {Object} [options] optional configuration options
+ * @param {string|undefined} [options.pathToOutputFile=undefined] where to write the openAPI specification to.
+ * Specify this to create the openAPI specification file.
+ * @param {number} [options.writeIntervalMs=10000] how often to write the openAPI specification to file
+ *
+ * @returns void
+ */
+function injectResponseMiddleware(expressApp, options = { pathToOutputFile: undefined, writeIntervalMs: 1000 * 10 }) {
```

Comment:
Shall we update documentation and add section like "Advanced usage" where usage of `injectRequestMiddleware/injectResponseMiddleware` is described.

Reply:
Yes, for sure.
What do you think about shorter names?
They look a little scary atm:D What about `handleRequests` and `handleResponses`? Or any other suggestions?

Reply:
Naming is the second biggest problem in IT. But having IDE with auto completition helps to deal with long names, so I see no problems in using long names.

Reply:
please add the section to README.md, I can polish it after if needed

Reply:
I wholeheartedly agree that naming this is very hard:D
I thought about renaming the functions to shorter names to allow the eye to catch the miss-placed order faster.

A great example is [the README I've just updated with Advanced usage](https://github.com/mpashkovskiy/express-oas-generator/pull/39/files#diff-04c6e90faac2675aa89e2176d2eec7d8). The words `inject` and `middleware` stay the same for both methods, and the word that matters the most is left off in the middle - either `request` or `response`.

Now, if the names were, say `handleResponses` and `handleRequests`, the word `middleware` wouldn't block off the important bit - `responses` and `requests` -- I think that this would be better in the UX part.

Also, shorter, but still perfectly self-explanatory variable names would mean that a lower cognitive load is required in the brain, which is also great:D